streamlit/ms2rescore_streamlit.py

```diff
@@ -0,0 +1,187 @@
+            "Percolator.",
+        )
+        self.user_input["ms2pip_model"] = st.sidebar.radio(
+            "MS²PIP model",
+            ["HCD", "CID", "TMT", "iTRAQ", "iTRAQphospho", "TTOF5600"],
+            help="MS²PIP model to use (see MS²PIP: https://github.com/compomics/ms2pip_c#ms-acquisition-information-and-peptide-properties-of-the-models-training-datasets)",
+        )
+        self.user_input["config_json"] = st.sidebar.text_area(
+            "Advanced JSON configuration",
+            value='{"general": {}}',
+            help="Set advanced MS²ReScore settings in JSON format (see https://github.com/compomics/ms2rescore/blob/master/configuration.md)",
+        )
+        st.sidebar.markdown(
+            """
+            See [MS²ReScore GitHub](https://github.com/compomics/ms2rescore/blob/master/configuration.md)
+            for all configuration details.
+            """
+        )
+
+    @staticmethod
+    def _parse_user_config(user_input: Dict, tmp_dir) -> Dict:
+        if not user_input["id_bytesio"]:
+            st.error("Please upload an identification file.")
+            return None
+        elif not user_input["mgf_bytesio"]:
+            st.error("Please upload an MGF file.")
+            return None
+        id_path = bytesio_to_tempfile(user_input["id_bytesio"])
+        mgf_path = bytesio_to_tempfile(user_input["mgf_bytesio"])
+        config_dict = _update_dict_recursively(
+            json.loads(user_input["config_json"]),
+            {
+                "general": {
+                    "identification_file": id_path,
+                    "mgf_path": mgf_path,
+                    "feature_sets": user_input["feature_sets"],
+                    "pipeline": user_input["pipeline"],
+                    "output_filename": os.path.join(
+                        tmp_dir.name, user_input["id_bytesio"].name
+                    ),
+                },
+                "ms2pip": {"model": user_input["ms2pip_model"]},
+            },
+        )
+        return config_dict
+
+    def _run_ms2rescore(self):
+        # Get config
+        tmp_dir = tempfile.TemporaryDirectory()
+        config_dict = self._parse_user_config(self.user_input, tmp_dir)
+        if not config_dict:
+            return None
+
+        # Run ms2rescore and send logs to front end
+        logger_placeholder = st.empty()
+        logger_name = "ms2rescore"
+        with StreamlitLogger(logger_placeholder, logger_name):
+            logging.info("Starting MS²ReScore...")
+            rescore = MS2ReScore(
+                parse_cli_args=False, configuration=config_dict, set_logger=False
+            )
+            rescore.run()
+
+        # Return results and cleanup files
+        st.markdown(
+            get_zipfile_href(
+                glob(config_dict["general"]["output_filename"] + "_*"),
+                filename="ms2rescore_results.zip",
+            ),
+            unsafe_allow_html=True,
+        )
+        shutil.rmtree(tmp_dir.name)
```

Comment:
You should use `tmp_dir.cleanup()` instead. Or use `tempfile.TemporaryDirectory` as a context manager.